### PR TITLE
feat(ui/picker): add props textFormatter

### PR DESCRIPTION
**Pull request by wangKBweb**
## Checklist

---

- [ ] Fix linting errors
- [ ] Tests have been added / updated (or snapshots)

## Change information

---

### CODE REVIEW

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 1]
```diff
@@ -71,6 +71,159 @@ const picker = async () => {
...
+
+const februaryDates = genCounts(28)
+
+const oddMonthDates = genCounts(31)
+
+const evenMonthDates = genCounts(30)
+
+function isBigMonth(month) {
+	return [1, 3, 5, 7, 8, 10, 12].includes(month)
+}
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/docs/en-US.md:95]: rename, format code

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 2]
```diff
@@ -71,6 +71,159 @@ const picker = async () => {
...
+    { label: '大地之灵', id: 3 },
+    { label: '虚空之灵', id: 4 },
+  ],
+  [
+    { label: '灰烬之灵', id: 1 },
+    { label: '风暴之灵', id: 2 },
+    { label: '大地之灵', id: 3 },
+    { label: '虚空之灵', id: 4 },
+  ],
+])
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/docs/en-US.md:189]: change to english

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 3]
```diff
@@ -71,6 +71,159 @@ const picker = async () => {
...
+function genDates(year, month) {
+  if (isLeapYear(year) && month === 2) {
+    return leapYearFebruaryDates
+  }
+
+  if (!isLeapYear(year) && month === 2) {
+    return februaryDates
+  }
+
+  if (isBigMonth(month)) {
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/docs/en-US.md:114]: rename

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 4]
```diff
@@ -71,6 +71,159 @@ const picker = async () => {
 </template>
 ```
 
+### textFormatter
+
+Picker passes in a `textFormatter` attribute to customize the text.
+`TextFormatter ` accepts two parameters. The first parameter `text` is the current text, and the second parameter  `columnIndex` is the subscript of the column where the current text is located.
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/docs/en-US.md:77]: textFormatter

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 5]
```diff
@@ -71,6 +71,159 @@ const picker = async () => {
...
+  if (isBigMonth(month)) {
+    return oddMonthDates
+  }
+
+  return evenMonthDates
+}
+  
+function genColumns(startYear, endYear) {
+  const columns = []
+  for (let i = startYear; i < endYear; i++) {
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/docs/en-US.md:123]: rename `i` to `year`

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 6]
```diff
@@ -71,6 +71,159 @@ const picker = async () => {
...
+const columns = computed(() => {
+  return serverColumns.value.map((column, index) => {
+    columnsMap.value[index] = {}
+    return column.map((item) => {
+      columnsMap.value[index][item.id] = item.label
+      return item.id
+    })
+  })
+})
+
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/docs/en-US.md:202]: too complicated

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 7]
```diff
@@ -132,6 +417,7 @@ const columns = ref(area)
 | `toolbar` | Whether to display the top toolbar | _string_ | `true` |
 | `cascade` | Whether to enable cascading mode | _boolean_ | `true` |
 | `cascade-initial-indexes` | List of initialization indices for cascade mode | _number[]_ | `-` |
+| `textFormatter` | textFormatter | _function_ | `text => text` |
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/docs/en-US.md:420]: text-formatter

**Comment by haoziqaq** [comment]
You should add `textFormatter` into `types/picker.d.ts`

**Comment by wegi8** [comment]
Maybe the `textFormatter` props can be a `string` | `()=> string`. Such as `textFormatter='YYYY-MM-DD HH:mm'`. like the `dayjs` or `momentjs`

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 8]
```diff
@@ -71,6 +71,149 @@ const picker = async () => {
...
+function genCounts(length) {
+  return Array.from({ length }, (_, index) => index + 1)
+}
+
+function isLeapYear(year) {
+  return (year % 4 === 0 && year % 100 !== 0) || year % 400 === 0
+}
+
+function genDates(year, month) {
+  if (isLeapYear(year) && month === 2) {
```
Comment (wegi8) [inline packages/varlet-ui/src/picker/docs/en-US.md:106]: I think is `February` more important than `isLeapYear`. Maybe we should to judge whether February isLeapYear first. If it's not `February`,This matter is meaningless.`

**Pull request merged**